src/python/WMComponent/DBS3Buffer/DBSUploadPoller.py

```diff
@@ -816,3 +822,16 @@ def checkBlocks(self):
 
         # We're done
         return
+
+    def getStepChainParentage(self, dataset):
+        myThread = threading.currentThread()
+        parentDatasetDAO = self.daoFactory(classname="UpdateBlocks")
```

Comment:
How about defining this DAO (and actually all the others) inside the __init__ method? And is it really going to be an update operation or just a select? Don't forget you need to create a new DAO for this.

In addition to that - there is no need to fix it in this PR - I just noticed we don't have the transaction handle in the algorithm method. In the last case it's handled in the BaseWorkerThread or something like that, but shouldn't we have the rollback and etc in algorithm?